src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -1948,35 +1951,180 @@ private String toString(TypeSpec typeSpec) {
-                    .addCode("return FOO;\n")
-                    .build())
-            .build();
-      assertThat(toString(taco)).isEqualTo(""
-            + "package com.squareup.tacos;\n"
-            + "\n"
-            + "import java.lang.Override;\n"
-            + "import java.lang.String;\n"
-            + "\n"
-            + "class Taco {\n"
-            + "  private static final String FOO;\n\n"
-            + "  static {\n"
-            + "    FOO = \"FOO\";\n"
-            + "  }\n\n"
-            + "  private String mFoo;\n\n"
-            + "  @Override\n"
-            + "  public String toString() {\n"
-            + "    return FOO;\n"
-            + "  }\n"
-            + "}\n");
+        .addField(String.class, "mFoo", Modifier.PRIVATE)
+        .addField(String.class, "FOO", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+        .addStaticBlock(CodeBlock.builder()
+            .addStatement("FOO = $S", "FOO")
+            .build())
+        .addMethod(MethodSpec.methodBuilder("toString")
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(String.class)
+            .addCode("return FOO;\n")
+            .build())
+        .build();
+    assertThat(toString(taco)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.lang.Override;\n"
+        + "import java.lang.String;\n"
+        + "\n"
+        + "class Taco {\n"
+        + "  private static final String FOO;\n\n"
+        + "  static {\n"
+        + "    FOO = \"FOO\";\n"
+        + "  }\n\n"
+        + "  private String mFoo;\n\n"
+        + "  @Override\n"
+        + "  public String toString() {\n"
+        + "    return FOO;\n"
+        + "  }\n"
+        + "}\n");
+  }
+
+  @Test public void initializerBlockInRightPlace() {
+    TypeSpec taco = TypeSpec.classBuilder("Taco")
+        .addField(String.class, "mFoo", Modifier.PRIVATE)
+        .addField(String.class, "FOO", Modifier.PRIVATE, Modifier.STATIC, Modifier.FINAL)
+        .addStaticBlock(CodeBlock.builder()
+            .addStatement("FOO = $S", "FOO")
+            .build())
+        .addMethod(MethodSpec.constructorBuilder().build())
+        .addMethod(MethodSpec.methodBuilder("toString")
+            .addAnnotation(Override.class)
+            .addModifiers(Modifier.PUBLIC)
+            .returns(String.class)
+            .addCode("return FOO;\n")
+            .build())
+        .addInitializerBlock(CodeBlock.builder()
+            .addStatement("mFoo = $S", "FOO")
+            .build())
+        .build();
+    assertThat(toString(taco)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.lang.Override;\n"
+        + "import java.lang.String;\n"
+        + "\n"
+        + "class Taco {\n"
+        + "  private static final String FOO;\n\n"
+        + "  static {\n"
+        + "    FOO = \"FOO\";\n"
+        + "  }\n\n"
+        + "  private String mFoo;\n\n"
```

Comment:
Agree! I have copy & pasted it from static block test. Thought that was desired by you guys ...